packages/main/src/DatePicker.ts

```diff
@@ -543,11 +567,14 @@ class DatePicker extends DateComponentBase implements IFormElement {
 
 	_updateValueState() {
 		const isValid = this._checkValueValidity(this.value);
+		const eventPrevented = !this.fireEvent<ValueStateChangeEventDetail>("value-state-change", { isValid }, true);
```

Comment:
The idea is not firing the event always, it's called value-state-change. It should be fired only when we are going to change the value-state (either feim bine to error, or from error to none).

Currently it is fired uncoditionally for example if someone selects 5 times a correct date, we will fire 5 times value-state-change. That's not matching the purpose and the name of the event.

The logic would have been correct if we are to introduce "validate" event or something like that.

Reply:
You're right! Thank you, I am correcting it.